From ca326311c5d7f45b71a416364e15d3655204cf29 Mon Sep 17 00:00:00 2001
From: Andrii Natiahlyi <natiahlyi@google.com>
Date: Thu, 21 Sep 2023 11:09:36 +0000
Subject: [PATCH] [M117] Fix PasswordManagerPorter is called after destruction

(cherry picked from commit 9dea1181c3af5209ba02fadf8e1a229b33b2ca6b)

Fixed: 1478889
Change-Id: I3acbf5dea56ea16a23a5799c241ca9aeb7ff091a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4872120
Auto-Submit: Andrii Natiahlyi <natiahlyi@google.com>
Commit-Queue: Vasilii Sukhanov <vasilii@chromium.org>
Reviewed-by: Vasilii Sukhanov <vasilii@chromium.org>
Commit-Queue: Andrii Natiahlyi <natiahlyi@google.com>
Cr-Original-Commit-Position: refs/heads/main@{#1197799}
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4882598
Cr-Commit-Position: refs/branch-heads/5938@{#1364}
Cr-Branched-From: 2b50cb4bcc2318034581a816714d9535dc38966d-refs/heads/main@{#1181205}
---

--- a/chrome/browser/ui/passwords/settings/password_manager_porter.cc
+++ b/chrome/browser/ui/passwords/settings/password_manager_porter.cc
@@ -70,7 +70,13 @@
       presenter_(presenter),
       on_export_progress_callback_(on_export_progress_callback) {}
 
-PasswordManagerPorter::~PasswordManagerPorter() = default;
+PasswordManagerPorter::~PasswordManagerPorter() {
+  // There may be open file selection dialogs. We need to let them know that we
+  // have gone away so that they do not attempt to call us back.
+  if (select_file_dialog_) {
+    select_file_dialog_->ListenerDestroyed();
+  }
+}
 
 bool PasswordManagerPorter::Export(content::WebContents* web_contents) {
   if (exporter_ && exporter_->GetProgressStatus() ==
